dev-notes/minimal-os-for-fobnail.md

```diff
@@ -0,0 +1,718 @@
+  ```
+
+Setup Zephyr build environment. Instructions below are based on Zephyr
+[Getting Started Guide](https://docs.zephyrproject.org/latest/develop/getting_started/index.html).
+
+* Install required packages, Ubuntu has too old packages so we have to use
+  KitWare repo.
+
+  ```shell
+  $ wget https://apt.kitware.com/kitware-archive.sh
+  $ sudo bash kitware-archive.sh
+  $ sudo apt install --no-install-recommends git cmake ninja-build gperf \
+    ccache dfu-util device-tree-compiler wget \
+    python3-dev python3-pip python3-setuptools python3-tk python3-wheel xz-utils file \
+    make gcc gcc-multilib g++-multilib libsdl2-dev python3-venv
+  $ pip3 install --user -U west
+  ```
+
+* Obtain Zephyr toolchain.
+
+  ```shell
+  $ wget https://github.com/zephyrproject-rtos/sdk-ng/releases/download/v0.14.1/zephyr-sdk-0.14.1_linux-x86_64.tar.gz
+  $ tar xf zephyr-sdk-0.14.1_linux-x86_64.tar.gz -C ~/.local
+  $ rm zephyr-sdk-0.14.1_linux-x86_64.tar.gz
+  $ ~/.local/zephyr-sdk-0.14.1/setup.sh -c
+  ```
+
+* Fetch Zephyr source code.
+
+  ```shell
+  $ west init zephyr
+  $ cd zephyr
+  $ west update
+  ```
+
+* Install required packages, use virtual environment to avoid conflict with
+  user/system packages.
+
+  ```shell
+  $ python3 -m venv .env
+  $ pip3 install -r zephyr/scripts/requirements.txt
+  ```
+
+* Build Zephyr, we use QEMU as the target, but it works on APU (without DLME).
+
+  ```shell
+  $ cd zephyr
+  $ west build -p auto -b qemu_x86_64 samples/userspace/hello_world_user/
+  $ sudo cp build/zephyr/zephyr.elf /mnt/boot/
+  ```
+
+* Configure GRUB, create '/mnt/boot/grub/grub.cfg` with the following contents.
+
+  ```shell
+  default 0
+  menuentry "Zephyr (DLME)" {
+    slaunch skinit
+    slaunch_module /skl.bin
+    multiboot /zephyr.elf
+  }
+  menuentry "Zephyr (no DLME)" {
+    multiboot /zephyr.elf
+  }
+  ```
+
+* Umount disk and sync before unplugging it.
+  > There were some problems with obtaining working binary, so before booting
+    Zephyr on target platform please check whether it boots in QEMU (without
+    DLME). When booting Zephyr you should see `Hello World` message on serial
+    console, if Zephyr doesn't output anything or if it stuck at
+    `Booting Zephyr` message then something went wrong.
+
+  ```shell
+  $ qemu-system-x86_64 -nographic -serial mon:stdio -m 512M -cpu qemu64 \
+    -drive format=raw,file=/dev/disk/by-id/usb-TS-RDF5_SD_Transcend_000000000039-0\:0
+  ```
+
+* Zephyr should normally boot on APU, if DLME variant (from boot menu) is used
+  Zephyr will also boot normally - that is because TrenchBoot GRUB executes SKL
+  only when using
+  [Multiboot2](https://github.com/3mdeb/grub/blob/tb_xen/grub-core/loader/multiboot.c#L192)
+  but Zephyr has support only for Multiboot1.
+
+* Booting Zephyr in DLME would require extending it with Multiboot2 support.
+
+## Summary
+
+* The above report outlines 8 operating systems that should be considered
+  candidates for Fobnail Token interoperability.
+
```

Comment:
I don't see point of empty line between bullets, I'm pretty sure it would look bad in mkdocs

Reply:
Fixed [0157aad](https://github.com/fobnail/docs/pull/26/commits/0157aad6fef7b0974587b919f0748211b5e7dc91#diff-76336fb0328e2063bb2c8b183eb9bcf50d8be89da43f108025b2c0624b8f0d6fL645)